### PR TITLE
ISPN-6333 - Adding more tests for Remote Script Execution

### PR DESCRIPTION
**Pull request by andyuk1986**
ISPN-6333 - https://issues.jboss.org/browse/ISPN-6333
- Added more tests verifying Script execution;
- Enabled a few disabled tests based on bugfixes; 

Fixed ISPN-6302 -  https://issues.jboss.org/browse/ISPN-6302

### CODE REVIEW

**Comment by slaskawi** [comment]
Hey @andyuk1986! Could you please rebase it? I would like to make sure we won't introduce new failures on CI.

**Comment by andyuk1986** [comment]
Hi @slaskawi :) Done!

File: server/integration/testsuite/src/test/java/org/infinispan/server/test/task/DistributedAuthServerTaskIT.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        String result = (String) remoteCache.execute(LocalAuthTestServerTask.NAME, Collections.emptyMap());
+        assertEquals(LocalAuthTestServerTask.EXECUTED_VALUE, result);
+        assertEquals(true, remoteCache.get(LocalAuthTestServerTask.KEY));
+
+        rcm.stop();
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void shouldRunDistAuthTest() throws Exception {
```
Comment (slaskawi) [inline server/integration/testsuite/src/test/java/org/infinispan/server/test/task/DistributedAuthServerTaskIT.java]: It seems there is a failure: http://ci.infinispan.org/viewLog.html?buildId=37568&buildTypeId=bt9&guest=1#testNameId-1517987404531832027

**Comment by slaskawi** [comment]
Thanks! It seems there is 1 related failure (I've checked the rest of them and they seem to be unrelated). The code looks good to me.

**Comment by andyuk1986** [comment]
@slaskawi I have fixed the test as well as all *ServerTaskIT tests by properly undeploying the deployed jar file. I have also reopend the DistributedServerTaskDeploymentIT test to see whether it will work with this solution. Waiting for CI build.

**Comment by andyuk1986** [comment]
@slaskawi I have fixed the failing test, as well as seems that DistributedServerTaskDeploymentIT test is not failing anymore - fixed with a separate commit. Also I have rebased with the latest changes. After that the testsuite was run twice on CI and no pull request related failures appeared.

**Comment by slaskawi** [comment]
Looks good to me.

**Comment by slaskawi** [comment]
Sorry, it's taking a bit long... Could you please rebase it once more?

**Comment by andyuk1986** [comment]
@slaskawi no problem :) Rebase is done :)

File: scripting/src/test/java/org/infinispan/scripting/ClusteredScriptingTest.java [hunk 2]
```diff
@@ -2,89 +2,233 @@
...
+   }
 
-      ArrayList<Map<String, Long>> resultsFuture =  (ArrayList<Map<String, Long>>) scriptingManager.runScript("wordCountStream_dist.js", new TaskContext().cache(cache(0))).get();
-      assertEquals(2, resultsFuture.size());
-      assertEquals(3209, resultsFuture.get(0).size());
-      assertEquals(3209, resultsFuture.get(1).size());
-      assertEquals(resultsFuture.get(0).get("macbeth"), Long.valueOf(287));
-      assertEquals(resultsFuture.get(1).get("macbeth"), Long.valueOf(287));
+   @DataProvider(name = "cacheModeProvider")
+   private static Object[][] providePrinciples() {
```
Comment (danberindei) [inline scripting/src/test/java/org/infinispan/scripting/ClusteredScriptingTest.java]: Method name seems out of place

Comment (andyuk1986) [inline scripting/src/test/java/org/infinispan/scripting/ClusteredScriptingTest.java]: Fixed!

**Comment by andyuk1986** [comment]
Rebase is done!

**Comment by slaskawi** [comment]
hmmm there seems to be a lot of failures: http://ci.infinispan.org/viewLog.html?buildId=37714&buildTypeId=bt9&guest=1

I retriggered CI...

**Comment by andyuk1986** [comment]
@slaskawi the only failure related to my pull request is: DistributedServerTaskIT.shouldThrowExceptionInRemoteTasks It is because of changes done by Will related to ISPN-6253. I will rebase once more my code and fix this test if it is not fixed already in the upstream. The CI build will be triggered again so we can watch if there is something which still fails.

**Comment by andyuk1986** [comment]
Rebased!

**Comment by andyuk1986** [comment]
@slaskawi the CI build is finished. There are 3 failures not related to the pull request.

**Comment by tristantarrant** [comment]
Thanks @andyuk1986 , pushed to master

**Pull request closed without merging**